internal/app/init/pkg/system/events/events.go

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package events
+
+import "time"
+
+// MaxEventsToKeep is maximum number of events to keep per service before dropping old entries
+const MaxEventsToKeep = 64
+
+// ServiceState is enum of service run states
+type ServiceState int
+
+// ServiceState constants
+const (
+	StateInitialized ServiceState = iota
+	StatePreparing
+	StateWaiting
+	StateRunning
+	StateStopping
+	StateFinished
+	StateFailed
+)
+
+func (state ServiceState) String() string {
+	switch state {
+	case StateInitialized:
+		return "Initialized"
+	case StatePreparing:
+		return "Preparing"
+	case StateWaiting:
+		return "Waiting"
+	case StateRunning:
+		return "Running"
+	case StateStopping:
+		return "Stopping"
+	case StateFinished:
+		return "Finished"
+	case StateFailed:
+		return "Failed"
+	default:
+		return "Unknown"
+	}
+}
+
+// ServiceEvent describes state change of the running service
+type ServiceEvent struct {
+	Message   string
+	State     ServiceState
+	Timestamp time.Time
+}
+
+// ServiceEvents is a fixed length history of events
+type ServiceEvents struct {
+	Events    []ServiceEvent
+	Discarded int
```

Comment:
Is Discarded just a counter? Should it be uint instead?

Reply:
yep, I thought it makes sense to know how many events were dropped. makes sense to make it uint for me, I will update it

Reply:
updated to `uint`